addons/test_mail/tests/common.py

```diff
@@ -12,6 +12,26 @@ class TestMailCommon(MailCommon):
     def _create_channel_listener(cls):
         cls.channel_listen = cls.env['mail.channel'].with_context(cls._test_context).create({'name': 'Listener'})
 
+    @classmethod
+    def _create_records_for_batch(cls, model, count):
+        records = cls.env[model]
+        partners = cls.env['res.partner']
+        country_id = cls.env.ref('base.be').id,
+        for x in range(count):
+            partners += cls.env['res.partner'].with_context(**cls._test_context).create({
+                'name': 'Partner_%s' % (x),
+                'email': '_test_partner_%s@example.com' % (x),
+                'country_id': country_id,
+                'mobile': '047500%02d%02d' % (x, x)
+            })
+            records += cls.env[model].with_context(**cls._test_context).create({
+                'name': 'Test_%s' % (x),
+                'customer_id': partners[x].id,
+            })
+        cls.records = cls._reset_mail_context(records)
+        cls.partners = partners
```

Comment:
Wow...
Can't you access partners from the created records when needed?

Or `return records, partners` and then assign them to "cls" if it's even necessary ?

This is a bit dirty / hidden.

Reply:
Well that's a move from other tests. I agree this is dirty and hidden but as it targets stable, I will keep it as it is, even for a test module.